frontend/src/pages/ActivityReport/Pages/components/Goal.js

```diff
@@ -11,11 +12,20 @@ import './Goal.css';
 const Goals = ({
   name, onRemoveGoal, goalIndex, objectives, onUpdateObjectives, createObjective,
 }) => {
+  const { unregister } = useFormContext();
   const onRemoveObjective = (index) => {
     const newObjectives = objectives.filter((o, objectiveIndex) => index !== objectiveIndex);
     onUpdateObjectives(newObjectives);
   };
 
+  const onLocalRemoveGoal = () => {
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < objectives.length; i++) {
+      unregister(`goals[${goalIndex}].objectives[${i}].ttaProvided`);
```

Comment:
The main issue that was happening was that the `RichEditor` still had a reference to the `objective` that was causing a bug in which the objective would persist even after deleting it.

Manually unregistering it was the fix.
Doing a for loop, for every objective that we currently have since a user might click `create a new objective` a million times, and thus creating a million references (and thus persist) so unregistering all of them will asure this bug won't happen again.